### PR TITLE
Handle functions and assignment in prefer-at condition

### PR DESCRIPTION
**Pull request by ShridharGoel**
https://github.com/Expensify/App/issues/43055

This is a follow-up to include handling of functions like map and assignment like `a[i] = 2`.

### CODE REVIEW

File: eslint-plugin-expensify/prefer-at.js [hunk 1]
```diff
@@ -67,12 +67,26 @@ module.exports = {
...
 
+                // Skip if the property is a method (like a.map)
+                if (node.parent && node.parent.type === AST_NODE_TYPES.CallExpression && node.parent.callee === node) {
+                    return;
+                }
+
+                // Skip if the node is part of an assignment expression (like a[i] = 2)
+                if (isAssignmentExpression(node)) {
+                    return;
+                }
```
[the remark this answers is not in the document]
Comment (eh2077) [reply eslint-plugin-expensify/prefer-at.js:88]: @ShridharGoel I did some search and found a similar implementation from this project [eslint-plugin-unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/5c92a9aed7793afa00f23e72e25bc58c10adecd4/rules/prefer-at.js#L142-L208) (under MIT license) and there's a [isLeftHandSide](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/5c92a9aed7793afa00f23e72e25bc58c10adecd4/rules/utils/is-left-hand-side.js) method. Maybe we can learn some ideas from there to make our implementation more robust.

**Comment by fabioh8010** [comment]
@ShridharGoel Could you please take a look at @eh2077 comments?

**Comment by fabioh8010** [comment]
@eh2077 Could you take a look now? Thanks 😄

**Comment by eh2077** [comment]
@ShridharGoel The unit test is failing. Can you please take a look?

**Comment by fabioh8010** [comment]
@eh2077 This unit test is already failing on `main` IIRC, I think we can ignore it.

File: eslint-plugin-expensify/utils/is-left-hand-side.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/rules/utils/is-left-hand-side.js
+
+function isLeftHandSide(node) {
+	return (
+		(node.parent.type === 'AssignmentExpression' || node.parent.type === 'AssignmentPattern')
```
Comment (eh2077) [inline eslint-plugin-expensify/utils/is-left-hand-side.js:5]: NAB: We use constants from 
```js
const { AST_NODE_TYPES, ESLintUtils } = require('@typescript-eslint/utils');
```
in `prefer-at.js`.

**Review by eh2077 — COMMENTED** [review]
LGTM! 

I ran test and lint locally but both of them failed though they're not caused by this change.

@fabioh8010 Can we just ignore them in this PR?

**Comment by fabioh8010** [comment]
Yes I would ignore them for now

**Review by eh2077 — APPROVED** [review]

**Review by roryabraham — APPROVED** [review]

**Pull request merged**